sdcm/remote/remote_file.py

```diff
@@ -23,12 +23,29 @@
 
 
 LOGGER = logging.getLogger(__name__)
+# The below set of SECURED_PARAMETERS should include relevant password-parameter-names,
+# to be masked by asterisks when printed to log.
+SECURED_PARAMETERS = {'ldap_bind_passwd'}
```

Comment:
This solution would work only on first level of keys, i.e. when there a password on a nested structure this would be enough (for example SSL configuration which is nested)

Reply:
@fruch , that is not correct (and also not the goal of this specific fix). it will hide the whole value including nested content, which might not be optimal, but does achieve the goal of secured logging..
just an example:
```
yaml_content
{'prepare_write_cmd': '   - >-\n    bin/ycsb load dynamodb -P workloads/workloadc -threads 80 -p recordcount=6990500\n    -p fieldcount=10 -p fieldlength=512 -p dataintegrity=true\n    -p insertstart=0 -p insertcount=3495250\n\n  - >-\n    bin/ycsb load dynamodb -P workloads/workloadc -threads 80 -p recordcount=6990500\n    -p fieldcount=10 -p fieldlength=512 -p dataintegrity=true\n    -p insertstart=3495250 -p insertcount=3495250\n\n  - >-\n    bin/ycsb load dynamodb -P workloads/workloadc -threads 80 -p recordcount=6990500\n    -p fieldcount=10 -p fieldlength=512 -p dataintegrity=true\n    -p insertstart=0 -p insertcount=3495250 -p table=usertable_no_lwt\n\n  - >-\n    bin/ycsb load dynamodb -P workloads/workloadc -threads 80 -p recordcount=6990500\n    -p fieldcount=10 -p fieldlength=512 -p dataintegrity=true\n    -p insertstart=3495250 -p insertcount=3495250 -p table=usertable_no_lwt '}

for key, value in yaml_content.items():
    if key in ['prepare_write_cmd']:
        yaml_content[key] = '*' * len(value)
        
yaml_content
{'prepare_write_cmd': '********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************'}
```

This fix was requested for the specific ldap parameter. If someone faces a different need in the future, it would be fine to adjust it.

Reply:
if I would want to hide `truststore_password`, you are saying I would need to hide all of `client_encryption_options` ?
```yaml
client_encryption_options:
  enabled: true
  keystore: <Path to keystore>/keystore.jks
  keystore_password: <keystore password> ## The password used when generating the keystore.
  truststore: <Path to keystore>/keystore.jks
  truststore_password: <keystore password>
  require_client_auth: <true or false>
```

Anyhow just wondering how hiding this from log, while we still upload the actual yaml into a public s3 bucket, is really keeping this password safe .... ?

Reply:
@fruch , i think one difference is that a password printed to log can be much easier searched through a github search or a similar, since it will be found on scylla issues details. But perhaps this S3 bucket is another security issue. @mykaul , please comment any advise per this discussion ( originated from https://github.com/scylladb/scylla-enterprise/issues/2678#issuecomment-1438670172 )